control-plane-operator/controllers/healthcheck/healthcheck_controller.go

```diff
@@ -0,0 +1,94 @@
+package healthcheck
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	hyperv1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/go-logr/logr"
+)
+
+const (
+	successRequeueInterval = 5 * time.Minute
+	failureRequeueInterval = 30 * time.Second
+)
+
+type HealthCheckUpdater struct {
+	client.Client
+	HostedControlPlane client.ObjectKey
+
+	log logr.Logger
+}
+
+func (hcu *HealthCheckUpdater) SetupWithManager(mgr ctrl.Manager) error {
+	return mgr.Add(hcu)
+}
+
+func (hcu *HealthCheckUpdater) Start(ctx context.Context) error {
+	hcu.log = ctrl.LoggerFrom(ctx).WithName("health-check-updater")
+	hcu.log.Info("Starting health check updater")
+	ticker := time.NewTicker(failureRequeueInterval)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-ticker.C:
+			err := hcu.update(ctx)
+			if err != nil {
+				hcu.log.Error(err, "Failure occurred during health checks")
+				ticker.Reset(failureRequeueInterval)
+			} else {
+				hcu.log.Info("Health checks succeeded")
+				ticker.Reset(successRequeueInterval)
+			}
+		}
+	}
+}
+
+func (hcu *HealthCheckUpdater) update(ctx context.Context) error {
+	hcu.log.Info("Updating health checks")
+
+	hostedControlPlane := &hyperv1.HostedControlPlane{}
+	err := hcu.Client.Get(ctx, hcu.HostedControlPlane, hostedControlPlane)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		return fmt.Errorf("failed to get HostedControlPlane: %w", err)
+	}
+	originalHostedControlPlane := hostedControlPlane.DeepCopy()
```

Comment:
should we abort here if the HCP is deleting?

Reply:
yes we should.  it wouldn't really cause an issue but would be messy on teardown (unable to update resource that is being deleted)